exec/java-exec/src/main/java/org/apache/drill/exec/record/AbstractBinaryRecordBatch.java

```diff
@@ -37,6 +38,30 @@
   // For now only used by Lateral and Merge Join
```

Comment:
This comment can be deleted (now that HJ and Union are about to use it too)

Reply:
done. Also, rebased and resolved merge conflicts. please look at latest diffs.